cloudify_agent/shell/commands/configure.py

```diff
@@ -49,8 +45,63 @@ def configure(disable_requiretty, relocated_env, no_sudo):
         click.echo('Disabling requiretty directive in sudoers file')
         _disable_requiretty(no_sudo)
         click.echo('Successfully disabled requiretty for cfy-agent')
-    if relocated_env:
-        _relocate_venv()
+    if fix_shebangs:
+        click.echo('Fixing shebangs in scripts')
+        _fixup_scripts()
+
+
+def _fixup_scripts():
+    """Make scripts in bin_dir relative by rewriting their shebangs
+
+    Examine each file in bin_dir - if it looks like a python script, and has a
+    shebang - replace it with a new, "relative" shebang.
+    """
+    from cloudify_agent.shell.main import get_logger
+    logger = get_logger()
+
+    new_shebang = f"#!{sys.executable}"
+    logger.debug('New shebang = %s', new_shebang)
+    for filename in _find_scripts_to_fix(os.path.dirname(sys.executable)):
+        logger.debug('Making script {0} relative'.format(filename))
+        _rewrite_shebang(filename, new_shebang)
+
+
+def _find_scripts_to_fix(bin_dir):
+    """Search bin_dir for files that look like python scripts with a shebang
+    """
+    for filename in os.listdir(bin_dir):
+        filename = os.path.join(bin_dir, filename)
+        if not os.path.isfile(filename):   # ignore subdirs, e.g. .svn ones
+            continue
+
+        with open(filename, 'rb') as f:
+            try:
+                lines = f.read().decode('utf-8').splitlines()
+            except UnicodeDecodeError:
+                # This is probably a binary program instead
+                # of a script, so just ignore it.
+                continue
+
+        if not lines:
+            continue
+
+        shebang = lines[0]
+        if not (shebang.startswith('#!') and 'bin/python' in shebang):
```

Comment:
This is broken by things like: #! /usr/bin/env python
I'm not sure whether we do care about that, though?

Reply:
it's not broken: `#!/usr/bin/env python` would be correct and wouldn't need changing by this (because it's not a hardcoded path to the python executable, and this code aims to fixup hardcoded paths to python executables).

Also, those files never contain `env python`, but they always contain hardcoded paths, and they pretty much have to: they need to run the virtualenv (or "virtualenv") python, even without being activated first